src/view/components/toolbar/motion/MotionSensorBar.tsx

```diff
@@ -0,0 +1,128 @@
+
+interface vscode {
+    postMessage(message: any): void;
+}
+
+declare const vscode: vscode;
+
+const sendMessage = (state: any) => {
+    vscode.postMessage({ command: "sensor-changed", text: state });
+};
+
+const MOTION_SLIDER_PROPS_X: ISliderProps = {
+    axisLabel: "X",
+    maxLabel: "Right",
+    maxValue: 78,
+    minLabel: "Left",
+    minValue: -78,
+    type: "motion_x",
+};
+const MOTION_SLIDER_PROPS_Y: ISliderProps = {
+    axisLabel: "Y",
+    maxLabel: "Front",
+    maxValue: 78,
+    minLabel: "Back",
+    minValue: -78,
+    type: "motion_y",
+};
+const MOTION_SLIDER_PROPS_Z: ISliderProps = {
+    axisLabel: "Z",
+    maxLabel: "Down",
+    maxValue: 78,
+    minLabel: "Up",
+    minValue: -78,
+    type: "motion_z",
+};
+
+const MOTION_SENSOR_PROPERTIES: ISensorProps = {
+    LABEL: "Motion sensor",
+    sliderProps: [
+        MOTION_SLIDER_PROPS_X,
+        MOTION_SLIDER_PROPS_Y,
+        MOTION_SLIDER_PROPS_Z,
+    ],
+    unitLabel: "Lux",
+};
+
+class MotionSensorBar extends React.Component {
+    constructor(props: any) {
+        super(props);
+    }
+
+    render() {
+        return (
+            <div className="MotionSensorBar">
+                <SensorButton
+                    label="Shake"
+                    type="shake"
+                    onMouseUp={this.onMouseUp}
+                    onMouseDown={this.onMouseDown}
+                    onKeyUp={this.onKeyUp}
+                    onKeyDown={this.onKeyDown}
+                />
+                <br />
+                <ThreeDimensionSlider
+                    axisProperties={MOTION_SENSOR_PROPERTIES}
+                />
+                <br />
+            </div>
+        );
+    }
+
+    private onMouseDown = (event: React.PointerEvent<HTMLButtonElement>) => {
+        this.updateShakePress(true, event.currentTarget.id);
+        this.handleOnclick(true, "shake");
+    };
+
+    private onKeyUp = (event: React.KeyboardEvent<HTMLButtonElement>) =>
+        this.onKeyEvent(event, false);
+
+    private onKeyDown = (event: React.KeyboardEvent<HTMLButtonElement>) =>
+        this.onKeyEvent(event, true);
+
+    private onMouseUp = (event: React.PointerEvent<HTMLButtonElement>) => {
+        this.updateShakePress(false, event.currentTarget.id);
+        this.handleOnclick(false, "shake");
+    };
+
+    private handleOnclick = (active: boolean, type: string) => {
+        const messageState = { [type]: active };
+        sendMessage(messageState);
+    };
+
+    private onKeyEvent(
+        event: React.KeyboardEvent<HTMLButtonElement>,
+        active: boolean
+    ) {
+        if (
+            [event.keyCode, event.key].includes(CONSTANTS.KEYBOARD_KEYS.ENTER)
+        ) {
+            this.handleOnclick(active, "shake");
+        }
+    }
+
+    private updateShakePress = (shakeState: boolean, id: string): void => {
+        const svgElement = window.document.getElementById("cpx_svg");
+        const buttonElement = window.document.getElementById(id);
```

Comment:
Is it possible to avoid this?

Reply:
Yes, I've only moved this file, the cpx still uses getelementbyid. I will do a separate pr for refactoring cpx specific code